cpp/tensorrt_llm/kernels/cutlass_kernels/allreduce_gemm/kernel/sm100_gemm_allreduce_tma_warpspecialized.hpp

```diff
@@ -418,7 +420,19 @@ class GemmARUniversal<ProblemShape_, CollectiveMainloop_, CollectiveEpilogue_, C
     CUTLASS_DEVICE
     void operator()(Params const& params, char* smem_buf)
     {
+        if constexpr (tensorrt_llm::kernels::arch::is_major_v<10>)
+        {
+            _invoke(params, smem_buf);
+        }
+        else
+        {
+            printf("ERROR : This kernel shall only run on SM10x devices.\n");
```

Comment:
Can we add an assert too so that with NDEBUG off we actually get a crash.
Also to make it easier to debug build issues in the future can we print the function name and file/line number using`__PRETTY_FUNCTION__` ` __FILE__` and `__LINE__` macros

Reply:
It would be good if we could perhaps have a way of checking this on the host side too to correctly propagate an error

Reply:
> This is device code so I want to keep the printf as simple as possible

It shouldn't be compiled in the `if constexpr` case right?

Can you do __PRETTY_FUNCTION__ instead of file? This way it should contain the relevant template instantiation as this may be relevant to debugging failures